runtime/sema/type.go

```diff
@@ -361,6 +361,46 @@ const toStringFunctionDocString = `
 A textual representation of this object
 `
 
+// fromString
+const FromStringFunctionName = "fromString"
+
+func FromStringMetadata(ty Type) (methodType *FunctionType, docString string) {
+	methodType = &FunctionType{
+		Parameters: []*Parameter{
+			{
+				Label:          ArgumentLabelNotRequired,
+				Identifier:     "input",
+				TypeAnnotation: NewTypeAnnotation(StringType),
+			},
+		},
+		ReturnTypeAnnotation: NewTypeAnnotation(
+			&OptionalType{ty},
+		),
+	}
+
+	builder := new(strings.Builder)
+	builder.WriteString(
+		fmt.Sprintf(
+			"Attempts to parse %s from a string. Returns `nil` on overflow or invalid input. Whitespace or invalid digits will return a nil value.\n",
+			ty.String(),
+		))
+
+	if IsSameTypeKind(ty, FixedPointType) {
```

Comment:
I'm taking a leap here and assuming that `IsSameTypeKind` is the correct check, instead of `IsSubType`.